app/javascript/components/App.js

```diff
@@ -26,9 +41,22 @@ class App extends React.Component {
           <Route path="/guidelines" component={CommunityGuidelinesPage} />
           <Route path="/contact" component={Contact} />
           <Route path="/protectedindex" component={ProtectedIndex} />
-          <Route path="/questionindex" component={QuestionIndex} />
+          <Route 
+            path="/questionindex"
+            render={props => <QuestionIndex user={this.props.current_user} 
+            questions={this.state.questions} />} 
+          />
           <Route path="/questionnew" component={QuestionNew} />
-          <Route path="/questionshow" component={QuestionShow} />
+          <Route 
+          path="/questionshow/:id"
+          render={props => {
+              let id = props.match.params.id
+              let question = this.state.questions.find((c) => c.id === +id)
+              console.log(question)
+              return <QuestionShow question={question} />
+              }
+              }
```

Comment:
The formatting is a bit off here. If you setup Prettier in VSCode, it should auto-format it on save which will fix this. Here's the docs for the package: https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode. Feel free to merge this if it's holding up other work, but make sure to do that. Having something autoformat code for you will lead to more consistent code.

Reply:
Thanks Shaun, we're running into dependency issues so we are trying to just push this working code. I will work with Shaun on adding the auto format on save.